ldcpy/dataarray_metrics.py

```diff
@@ -0,0 +1,329 @@
+import collections
+import copy
+import typing
+
+import dask
+import numpy as np
+import pandas as pd
+import scipy.stats
+import xarray as xr
+import xrft
+
+xr.set_options(keep_attrs=True)
+
+
+@xr.register_dataarray_accessor('ldc')
+class MetricsAccessor:
+    """
+    This class contains metrics for each point of a dataset after aggregating across one or more dimensions, and a method to access these metrics.
+    """
+
+    def __init__(self, xarray_obj):
+        self._obj = xarray_obj.copy()
+        self._attrs = self._obj.attrs.copy()
+        self._metrics = self._init_metrics()
+        self._aggregate_dims = None
+        self._is_computed = False
+
+    def _init_metrics(self):
+        keys = [
+            'ns_con_var',
+            'ew_con_var',
+            'mean_val',
+            'mean_abs',
+            'standard_deviation',
+            'prob_positive',
+            'odds_positive',
+            'prob_negative',
+            'zscore',
+            'mae_day_max',
+            'lag1',
+            'lag1_first_difference',
+            'quantile_val',
+            'mean_squared',
+            'root_mean_squared',
+            'sum_val',
+            'sum_squared',
+            'variance',
+            'pooled_variance',
+            'pooled_variance_ratio',
+            'standardized_mean',
+            'max_abs',
+            'max_abs',
+            'min_abs',
+            'dyn_range',
+            'min_val',
+            'max_val',
+            'annual_harmonic_relative_ratio',
+            'annual_harmonic_relative_ratio_pct_sig',
+            'zscore_cutoff',
+            'zscore_percent_significant',
+        ]
+        return pd.Series(collections.OrderedDict.fromkeys(sorted(keys)))
+
+    def __call__(
+        self,
+        aggregate_dims: typing.Union[typing.List, typing.Tuple],
+        grouping: typing.Union[typing.List, typing.Tuple] = None,
+        ddof: int = 1,
+        q: float = 0.5,
+        spre_tol: float = 1.0e-4,
+        time_dim_name: str = 'time',
+        lat_dim_name: str = 'lat',
+        lon_dim_name: str = 'lon',
```

Comment:
Also, I tried removing the hard-coded dimensions names by making them configurable.